src/networkProvider/proxyNetworkProvider.ts

```diff
@@ -0,0 +1,203 @@
+        let response = await this.doGetGeneric(url);
+        let transaction = TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction);
+        return transaction;
+    }
+
+    async getTransactionStatus(txHash: TransactionHash): Promise<TransactionStatus> {
+        let response = await this.doGetGeneric(`transaction/${txHash.toString()}/status`);
+        let status = new TransactionStatus(response.status);
+        return status;
+    }
+
+    async sendTransaction(tx: Transaction): Promise<TransactionHash> {
+        let response = await this.doPostGeneric("transaction/send", tx.toSendable());
+        let hash = new TransactionHash(response.txHash);
+        return hash;
+    }
+
+    async simulateTransaction(tx: Transaction): Promise<any> {
+        let response = await this.doPostGeneric("transaction/simulate", tx.toSendable());
+        return response;
+    }
+
+    async queryContract(query: Query): Promise<IContractQueryResponse> {
+        let data = query.toHttpRequest();
+        let response = await this.doPostGeneric("vm-values/query", data);
+        let queryResponse = ContractQueryResponse.fromHttpResponse(response.data);
+        return queryResponse;
+    }
+
+    async getDefinitionOfFungibleToken(_tokenIdentifier: string): Promise<IDefinitionOfFungibleTokenOnNetwork> {
+        // TODO: Implement wrt.:
+        // https://github.com/ElrondNetwork/api.elrond.com/blob/main/src/endpoints/esdt/esdt.service.ts#L221
+        throw new Error("Method not implemented.");
+    }
+
+    async getDefinitionOfTokenCollection(_collection: string): Promise<IDefinitionOfTokenCollectionOnNetwork> {
+        // TODO: Implement wrt.:
+        // https://github.com/ElrondNetwork/api.elrond.com/blob/main/src/endpoints/collections/collection.service.ts
+        // https://docs.elrond.com/developers/esdt-tokens/#get-esdt-token-properties
+        throw new Error("Method not implemented.");
+    }
+
+    async getNonFungibleToken(_collection: string, _nonce: Nonce): Promise<INonFungibleTokenOfAccountOnNetwork> {
+        throw new Error("Method not implemented.");
+    }
+
+    async doGetGeneric(resourceUrl: string): Promise<any> {
+        let response = await this.doGet(resourceUrl);
+        return response;
+    }
+
+    async doPostGeneric(resourceUrl: string, payload: any): Promise<any> {
+        let response = await this.doPost(resourceUrl, payload);
+        return response;
+    }
+
+    private async doGet(resourceUrl: string): Promise<any> {
+        try {
+            let url = `${this.url}/${resourceUrl}`;
+            let response = await axios.get(url, this.config);
+            let payload = response.data.data;
+            return payload;
+        } catch (error) {
+            this.handleApiError(error, resourceUrl);
+        }
+    }
+
+    private async doPost(resourceUrl: string, payload: any): Promise<any> {
```

Comment:
Should we have a TODO or something to also remove duplicate code between these providers?

Reply:
While there is symmetry between the two providers, most functions are different (due to response enveloping, types of fields, URLs etc.). However, added TODOs (there are some duplicated chunks).